meta-balena-common/recipes-core/balena-rollback/files/rollback-tests

```diff
@@ -40,3 +40,15 @@ else
 	echo "Rollback: ERROR: balenaEngine is not healthy!"
 	exit 1
 fi
+
+# The following healthcheck is installed only on devices
+# which support it, and is provided by the device integration layer.
+if [ -f /usr/bin/jetson-firmware-healthcheck ]; then
+	echo "Rollback: Jetson UEFI firmware used, checking if firmware has been updated..."
+	if /usr/bin/jetson-firmware-healthcheck ; then
+		echo "Rollback: Jetson UEFI Firmware L4T version matches Kernel L4T version"
+	else
+		echo "Rollback: ERROR: Jetson UEFI Firmware L4T version does not match Kernel L4T version!"
+		exit 1
+	fi
+fi
```

Comment:
hey @acostach could we have this more generic? Like, let's add a hook where device integration layers can add a healthcheck, not a Jetson specific one.

Reply:
hey @acostach the first option is what I was thinking about too - no need to complicate it with new breadcrumbs I think

Reply:
Thanks Alex! I updated the PR